source/MaterialXGenGlsl/GlslShaderGenerator.cpp

```diff
@@ -666,7 +666,8 @@ void GlslShaderGenerator::emitVariableDeclaration(const ShaderPort* variable, co
     if (variable->getType() == Type::FILENAME)
     {
         // Samplers must always be uniforms
-        emitString("uniform sampler2D " + variable->getVariable(), stage);
+        string str = qualifier.empty() ? EMPTY_STRING : qualifier + " ";
+        emitString(str + "sampler2D " + variable->getVariable(), stage);
```

Comment:
When called in a parameter block context, the value of `qualifier` will be "uniform". When called in a function parameter context, the `qualifier` will be the empty string.